packages/server/lib/util/trash.coffee

```diff
@@ -1,8 +1,11 @@
+fs      = require("fs")
 path    = require("path")
 trash   = require("trash")
 Promise = require("bluebird")
 
 module.exports = {
   folder: (pathToFolder) ->
-    Promise.resolve(trash([path.join(pathToFolder, "**"),"!#{pathToFolder}"]))
+    Promise.all(fs.readdirSync(pathToFolder).map (item) ->
```

Comment:
We can't have sync fs calls in the server because they can cause `EMFILE` errors that `graceful-fs` can't catch and handle. `packages/server` has a promisified version of `fs-extra` under `lib/util/fs`, so you can use `readdirAsync` from there instead. 

I also think this is causing the failing tests because it throws when the directory being read doesn't exist.